packages/story-editor/src/components/panels/design/textStyle/stylePresets/stylePresets.js

```diff
@@ -0,0 +1,188 @@
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import {
+  THEME_CONSTANTS,
+  Text,
+  BUTTON_TYPES,
+  BUTTON_SIZES,
+  BUTTON_VARIANTS,
+  Icons,
+  Button,
+  PLACEMENT,
+} from '@web-stories-wp/design-system';
+import { __ } from '@web-stories-wp/i18n';
+import styled from 'styled-components';
+import { useMemo, useRef, useState } from '@web-stories-wp/react';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../../../../app';
+import { PRESET_TYPES } from '../../../../../constants';
+import useAddPreset from '../../../../../utils/useAddPreset';
+import { focusStyle } from '../../../shared';
+import Popup from '../../../../popup';
+import useInspector from '../../../../inspector/useInspector';
+import useApplyStyle from './useApplyStyle';
+import StyleGroup from './styleGroup';
+import StyleManager from './styleManager';
+
+const PresetsHeader = styled.div`
+  display: flex;
+  padding: 8px 0;
+  justify-content: space-between;
+`;
+
+const StylesWrapper = styled.div``;
+
+const SubHeading = styled(Text)`
+  color: ${({ theme }) => theme.colors.fg.secondary};
+  margin: 6px 0;
+`;
+
+const StyledButton = styled(Button)`
+  margin-left: auto;
+`;
+
+const StyledMoreButton = styled(Button)`
+  ${focusStyle};
+  margin: 12px 0;
+  padding: 0 16px;
+  justify-content: center;
+  align-self: center;
+  svg {
+    transform: rotate(-90deg);
+    height: 32px;
+    width: 32px;
+    margin-left: -4px;
+  }
+`;
+
+const NoStylesWrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: calc(100% + 32px);
+  margin-left: -16px;
+  background-color: ${({ theme }) => theme.colors.opacity.black24};
+  height: 64px;
+`;
+
+const NoStylesText = styled(Text)`
+  color: ${({ theme }) => theme.colors.fg.tertiary};
+`;
+
+function PresetPanel({ pushUpdate }) {
+  const { globalStoryStyles } = useStory(
+    ({
+      state: {
+        story: { globalStoryStyles },
+      },
+    }) => {
+      return {
+        globalStoryStyles,
+      };
+    }
+  );
+
+  const {
+    refs: { inspector },
+  } = useInspector();
+  const buttonRef = useRef(null);
+  const [isPopupOpen, setIsPopupOpen] = useState(false);
+
+  const { textStyles } = globalStoryStyles;
+  const hasPresets = textStyles.length > 0;
+
+  const handleApplyStyle = useApplyStyle({ pushUpdate });
+  const { addGlobalPreset } = useAddPreset({ presetType: PRESET_TYPES.STYLE });
+
+  const handlePresetClick = (preset) => {
+    handleApplyStyle(preset);
+  };
+
+  const spacing = useMemo(() => ({ x: 20 }), []);
```

Comment:
Does it make a difference to memoize this? Is the popup memoized thus requiring the same argument (same object) every time? Since it's static anyway, you could just move it to a `const` outside the component.